From c69410a78666853454fafca071dea49fb4a4c515 Mon Sep 17 00:00:00 2001
From: Tomas Mraz <tomas@openssl.org>
Date: Mon, 1 Aug 2022 13:08:45 +0200
Subject: [PATCH 098/209] Properly document deprecation of DH_new() and related
 functions

Reviewed-by: Matt Caswell <matt@openssl.org>
Reviewed-by: Todd Short <todd.short@me.com>
Reviewed-by: Hugo Landau <hlandau@openssl.org>
(Merged from https://github.com/openssl/openssl/pull/18925)

(cherry picked from commit 65b41ab31917e3bb25920c4b27017786db668850)
---
 doc/man3/DH_new.pod        | 13 ++++++++++++-
 doc/man3/DH_new_by_nid.pod |  5 +++--
 2 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/doc/man3/DH_new.pod b/doc/man3/DH_new.pod
index 62c61c3d39..dc073b4e3f 100644
--- a/doc/man3/DH_new.pod
+++ b/doc/man3/DH_new.pod
@@ -8,6 +8,10 @@ DH_new, DH_free - allocate and free DH objects
 
  #include <openssl/dh.h>
 
+The following functions have been deprecated since OpenSSL 3.0, and can be
+hidden entirely by defining B<OPENSSL_API_COMPAT> with a suitable version value,
+see L<openssl_user_macros(7)>:
+
  DH* DH_new(void);
 
  void DH_free(DH *dh);
@@ -32,7 +36,14 @@ DH_free() returns no value.
 
 L<DH_new(3)>, L<ERR_get_error(3)>,
 L<DH_generate_parameters(3)>,
-L<DH_generate_key(3)>
+L<DH_generate_key(3)>,
+L<EVP_PKEY-DH(7)>
+
+=head1 HISTORY
+
+All of these functions were deprecated in OpenSSL 3.0.
+
+For replacement see EVP_PKEY-DH(7).
 
 =head1 COPYRIGHT
 
diff --git a/doc/man3/DH_new_by_nid.pod b/doc/man3/DH_new_by_nid.pod
index d5ad0ff6ce..808205f196 100644
--- a/doc/man3/DH_new_by_nid.pod
+++ b/doc/man3/DH_new_by_nid.pod
@@ -7,12 +7,13 @@ DH_new_by_nid, DH_get_nid - create or get DH named parameters
 =head1 SYNOPSIS
 
  #include <openssl/dh.h>
- DH *DH_new_by_nid(int nid);
 
 The following functions have been deprecated since OpenSSL 3.0, and can be
 hidden entirely by defining B<OPENSSL_API_COMPAT> with a suitable version value,
 see L<openssl_user_macros(7)>:
 
+ DH *DH_new_by_nid(int nid);
+
  int DH_get_nid(const DH *dh);
 
 =head1 DESCRIPTION
@@ -37,7 +38,7 @@ and optionally q, otherwise it returns B<NID_undef> if there is no match.
 
 =head1 HISTORY
 
-The DH_get_nid() function was deprecated in OpenSSL 3.0.
+All of these functions were deprecated in OpenSSL 3.0.
 
 =head1 COPYRIGHT
 
-- 
2.34.1

